src/shared/dissect-image.c

```diff
@@ -1112,9 +1112,14 @@ DissectedImage* dissected_image_unref(DissectedImage *m) {
         if (!m)
                 return NULL;
 
+        /* First, clear dissected partitions. */
         for (PartitionDesignator i = 0; i < _PARTITION_DESIGNATOR_MAX; i++)
                 dissected_partition_done(m->partitions + i);
 
+        /* Second, free decrypted images. This must be after dissected_partition_done(), as freeing
+         * DecryptedImage may try to deactivate partitions. */
+        decrypted_image_unref(m->decrypted_image);
```

Comment:
I'm not sure about this, all that `dissected_partition_done()` does is free some values, why is it so important to do `decrypted_image_unref()` after that?

Reply:
If a decrypted partition is deactivated, then the contents of corresponding DissectedPartition object may be outdated, especially `DissectedPartition.decrypted_node`. To make not leave dirty information, we need to free DissectedPartition at first.

Of course, with the current code, the order of these steps can be arbitrary, but for safety.